src/constructs/Queue.ts

```diff
@@ -155,6 +160,8 @@ export class Queue extends AwsConstruct {
         return {
             queueUrl: this.referenceQueueUrl(),
             queueArn: this.referenceQueueArn(),
+            dlqUrl: this.getDlqUrl(),
+            dlqArn: thus.getDlqArn(),
```

Comment:
you can't use those functions as they use the Cloudformation stack outputs. But you need the references (Cloudformation's `Ref` or `GetAtt`) to be able to reference it custom cloudformation, function environment variables, event triggers on function etc

That's the purpose of `AwsProvider::getCloudFormationReference`. This function will use the CDK to resolve references which in the end uses CDK Tokens IIRC. 

Check `referenceQueueUrl` to get an idea of how to do it here.